internal/client/client.go

```diff
@@ -105,10 +105,15 @@ type Client struct {
 	PartitionService     *icluster.PartitionService
 	ClusterService       *icluster.Service
 	name                 string
-	shutdownHandlers     []ShutdownHandler
+	shutdownHandlers     shutdownHandler
 	state                int32
 }
 
+func (c *Client) AddShutdownHandler(handler ...func(ctx context.Context)) {
```

Comment:
Do we need this change?

Reply:
Resolved, [0042502](https://github.com/hazelcast/hazelcast-go-client/pull/772/commits/0042502d165af4bbfd57b99474f816dd147c4599).